conda_forge_tick/migrators/extra_jinj2a_keys_cleanup.py

```diff
@@ -0,0 +1,67 @@
+import os
+import typing
+from typing import Any
+from ruamel.yaml import YAML
+
+from conda_forge_tick.xonsh_utils import indir
+from conda_forge_tick.migrators.core import MiniMigrator
+from conda_forge_tick.recipe_parser import CondaMetaYAML
+
+if typing.TYPE_CHECKING:
+    from ..migrators_types import AttrsTypedDict
+
+
+class ExtraJinja2KeysCleanup(MiniMigrator):
+    """Remove extra Jinja2 keys that make things hard for the version migrator.
+
+    Variables removed:
+
+    - ``hash_type``: This is sometimes used as a key in the ``source`` section
+      to map the hash_type (ex. sha256) to the hash.
+    - ``hash_value``: This is unnecessary since it is typically only used once
+      in a YAML recipe.
+    - ``file_ext``: This is unnecessary and just makes things harder to read.
+      It may also be causing trouble for the version migrator.
```

Comment:
Should this be done as a migrator or something that the linter complains about?

Reply:
You have the migrator so it is fine :) But should we also prevent this from happening in the future with linter warnings?

Reply:
@beckermr and I had talked about making the auto tick bot create an issue if it fails to update a recipe. This was happening for some of my recipes where a new version came out but the feedstock showed nothing (no PR, no issue) from the bots.

The main unknown for me on linter warnings is I'm not exactly sure *why* this was a problem for the version migrator, just that it was. I think not allowing jinja2 variables as YAML keys would be an easy lint check to do. The hash_value and file_ext are just extra.

Reply:
Sounds good!